pepper/apps/dex_common.h

```diff
@@ -34,6 +34,16 @@ uint32_t sumBits(bool *bits, uint32_t length) {
     return result;
 }
 
+/**
+ * checks that all numbers in the array are of boolean type  
+ */
+void isBoolVerification(bool *bits, uint32_t length) {
+    uint32_t index = 0;
+    for (index=0; index<length; index++) {
+        assert_zero(bits[length-1-index] * bits[length-1-index] -bits[length-1-index] );
```

Comment:
Why are you going backwards? Wouldn't bits[index] also work? It seems a bit more concise and straightforward.
Could you check what the impact in terms of extra constraints is (e.g. by running the e2e test for 2 orders).
I wonder if it would be cheaper to test assert_zero(bits[index]*(bits[index]-1))

Also, could you please add a unit test in C (for both the valid and invalid case)?